src/provisioning-ofono.c

```diff
@@ -1230,10 +1278,9 @@ int provisioning_init_ofono(void)
  * Currently this assumes that only one modem in ofono if need to support more
  * it could be something like below
  */
-#ifdef MULTIMODEM
 	modem_list = g_hash_table_new_full(g_str_hash, g_str_equal,
-						NULL, remove_modem);
-#endif
+						NULL, remove_modems);
+
```

Comment:
Sounds like you are heading in the wrong direction. There must be a piece of code that cleans things up after EACH modem. It should be called whenever the modem gets removed from the table. If you have more than one modem, it will be called more than once, there's nothing wrong with that. It's like a C++ destructor getting called as many times as many objects you destroy. This is really basic stuff.